explainer.md

```diff
@@ -22,118 +71,258 @@ place associated with the application (such as the shelf or home screen).
+
+The main page of our site https://example.com/
+```html
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <title>Main Page</title>
+  </head>
+  <body>
+    <script>
+      // There are ten unread notifications.
+      // This was injected by the server.
+      Badge.set(10);
+    </script>
+  </body>
+</html>
+```
+
+On https://example.com/do-i-have-https
+```html
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <title>Main Page</title>
+  </head>
+  <body>
+    <script>
+      const badgeOptions = {
+        scope: `/do-i-have-https`
+      };
+
+      const hasHttps = window
+        .location
+        .href
+        .startsWith('https');
+
+      // But we could do this instead:
+      if (hasHttps)
+        Badge.set(badgeOptions);
+      else Badge.clear(badgeOptions);
+
+    </script>
+  </body>
+</html>
+```
+
+##### Badging for Multiple Apps on the Same Origin (as in the case of multiple Github Pages PWAs)
+```js
+// Scope of Frobnicate is /frobnicate
+// Scope of Bazify is /baz
+// Both apps are hosted on https://origin.example/${scope}
+
+// By default, setting a badge affects the origin.
+Badge.set();
+// Badge for Frobnicate and Bazify is 'flag'.
+// Badge for /index.html is 'flag'.
+
+// Installed applications use the badge with the most
+// specific scope which encompasses their own scope.
+Badge.set(7, { scope: `/baz` });
+// Badge for Frobnicate is 'flag'.
+// Badge for Bazify is '7'.
+// Badge for /index.html is 'flag'.
+
+// Pages inside an installed application do not effect
+// the badge displayed in an OS specific context for
+// that app.
+Badge.set(55, { scope: `/frobnicate/page` });
+// Badge for Frobnicate is 'flag'.
+// Badge for Bazify is '7'.
+// Badge for /index.html is 'flag'.
+// Badge for /frobnicate/page is '55'.
+
+// Setting the badge for '/' is equivalent to omitting
+// the scope param.
+Badge.set(88, { scope: `/` });
+// Badge for Frobnicate is '88'.
+// Badge for Bazify is '7'.
+// Badge for /index.html is '88'.
+// Badge for /frobnicate/page is '55'.
+
+// Clearing the badge for a scope will cause the next
+// most specific badge to be applied.
+Badge.clear({ scope: '/baz' });
+// Badge for Frobnicate is '88'.
+// Badge for Bazify is '88'.
+// Badge for /index.html is '88'.
+// Badge for /frobnicate/page is '55'.
+
+// Clears the badge for the origin.
+Badge.clear();
+// Badge for Frobnicate, Bazify, /index.html is nothing.
+// Badge for /frobnicate/page is '55'.
+```
+
 ## UX treatment
+Badges should appear as a user agent defined overlay on top of the app's icon, about a quarter of the size, and in one of the four corners. If the exact representation of a badge is not supported (e.g., a 2-digit number while only single characters are allowed, or a character where only numbers are allowed) the user agent should make a best effort attempt to map the unsupported data onto supported data. This may involve:
```

Comment:
Reword this section so it's neutral as to whether we're doing app badging or tab badging. (Instead of "badges go over the app's icon, and may also go on the tab", say "badges can go in any place the user agent feels appropriate to indicate status, including the app icon or the page icon in a browser tab" (go into more detail than what I have here).